### ISSUE TITLE
Success icon missing 

### ISSUE THREAD

**Issue by RiodeJaneiroo** [emilkowalski/sonner#349]
### Describe the feature / bug 📝:

When using toast.promise, the success icon is not displayed after the promise resolves. This behavior persists across different environments, including on the official documentation site.

### Steps to reproduce the bug 🔁:
it also do not display on sonner site)
[Codesandbox](https://codesandbox.io/p/sandbox/dank-darkness-kf9qs7)

**Comment by combdn** [comment]
I'm unsure if it's related or a separate issue, but if I define my own icons, the spinner is never replaced by the success icon.

https://github.com/emilkowalski/sonner/assets/889260/c276febf-fdab-480b-8903-49243dbd317c

**Comment by emilkowalski** [comment]
Fixed all bugs reported here in #352

**Comment by combdn** [comment]
In 1.4.2, my spinner is showing next to the success icon:

https://github.com/emilkowalski/sonner/assets/889260/f5b49ac0-3adf-420d-888c-cad539213a95

Icons configuration:
```
{
  success: <BsCheckCircleFill />,
  info: <BsInfoCircleFill />,
  warning: <BsExclamationTriangleFill />,
  error: <BsExclamationOctagonFill />,
  loading: <Spinner />,
}
```
If I remove my spinner icon, it works correctly (shows the spinner, then the spinner is replaced by the success icon).
